src/Traits/HasPermissions.php

```diff
@@ -101,6 +135,15 @@ protected function convertToPermissionModels($permissions): array
         }, $permissions);
     }
 
+    protected function convertPermissionsToRoleIds($permissions)
```

Comment:
@drbyte I haven't had the chance to test this in production yet so nothing concrete to report.  In theory, introducing  `$query->pluck('id')` in this method (where `array_column('id', $roles)` was previously used in the calling method to calculate permission ids for new/existing scopes) could reduce memory usage.